scripts/power_test.py

```diff
@@ -1,54 +1,143 @@
-#!/usr/bin/env python
-import pyudev
-from power import Power
+#!/usr/bin/env python3
+
+"""
+Power board test script.
+
+This will:
+
+ - hopefully not crash
+"""
+
 import time
+from typing import Any, Iterable, NamedTuple, Optional
+
+import usb.core
+import usb.util
+
+class USBDevice(NamedTuple):
+    """A single device found on USB."""
+
+    serial: str
+    node: usb.core.Device
+    busnum: int
+    devnum: int
+
+
+def _enumerate_usb_devices(*, product: int, vendor: int) -> Iterable[USBDevice]:
+    """Create a sorted list of USB devices of the given type."""
+    devices = list(usb.core.find(find_all=True, idProduct=product, idVendor=vendor))
+    #devices.sort(key=lambda x: x.serial_number)
+    for device in devices:
+        yield USBDevice(
+            serial=0,#device.serial_number,
+            busnum=device.bus,
+            devnum=device.address,
+            node=device,
+        )
+
+class PowerBoard(object):
+    """A single power board interface (backed by a USBDevice)."""
+
+    def __init__(self, device: USBDevice) -> None:
+        """Construct power board from `USBDevice`."""
+        self.device = device
+        self._configuration: Optional[usb.core.Endpoint]
```

Comment:
I would enjoy this being compatible with Python 3.5 - remove this line and type annotate via a comment on the next line (add `  # type: Optional[usb.core.Endpoint]` there)